core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -331,7 +331,12 @@ private[spark] class MemoryStore(
     var unrollMemoryUsedByThisBlock = 0L
     // Underlying buffer for unrolling the block
     val redirectableStream = new RedirectableOutputStream
-    val bbos = new ChunkedByteBufferOutputStream(initialMemoryThreshold.toInt, allocator)
+    val chunkSize = if (initialMemoryThreshold > Int.MaxValue) {
+      4 * 1024 * 1024
```

Comment:
Because `initialMemoryThreshold` is set to `unrollMemoryThreshold` which is configurable, if we want to use a default value in this case, we should log a warning message at least.

Another option is just throwing an exception. I am ok for either one.